src/flb_utils.c

```diff
@@ -1303,7 +1303,9 @@ int flb_utils_read_file(char *path, char **out_buf, size_t *out_size)
 
     bytes = fread(buf, st.st_size, 1, fp);
     if (bytes < 1) {
-        flb_errno();
+        if (bytes < 0) {
```

Comment:
How about using `ferror` ?
https://man7.org/linux/man-pages/man3/ferror.3.html

Reply:
I'll add that. Did not even know about that function.

Reply:
interesting, it seems ferror() is the way to go.